wetterdienst/provider/noaa/ghcn/api.py

```diff
@@ -0,0 +1,325 @@
+        # store parameter names (long accessors)
+        tmax_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MAX_200.value
+        tmin_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MIN_200.value
+        tmean_key = NoaaGhcnParameter.DAILY.TEMPERATURE_AIR_MEAN_200.value
+
+        drop_columns = [
+            Columns.STATION_ID.value,
+            Columns.PARAMETER.value,
+            Columns.QUALITY.value,
+        ]
+
+        # TODO: better manage start and end date and its timezone
+        # code taken from ValuesCore._complete_dates
+        timezone_ = self._get_timezone_from_station(station_id)
+
+        # draw tmax and tmin data from DataFrame
+        df_tmax = (
+            df[df[Columns.PARAMETER.value] == tmax_key]
+            .drop(columns=drop_columns)
+            .rename(columns={Columns.VALUE.value: tmax_key})
+        )
+        df_tmin = (
+            df[df[Columns.PARAMETER.value] == tmin_key]
+            .drop(columns=drop_columns)
+            .rename(columns={Columns.VALUE.value: tmin_key})
+        )
+
+        start_date = self.stations.start_date
+        end_date = self.stations.end_date
+
+        if not start_date:
+            # either take start and end date from temperature dfs
+            if not df_tmax.empty or not df_tmin.empty:
+                start_date = np.nanmin(
+                    [
+                        df_tmax[Columns.DATE.value].min(),
+                        df_tmin[Columns.DATE.value].min(),
+                    ]
+                )
+                end_date = np.nanmax(
+                    [
+                        df_tmax[Columns.DATE.value].max(),
+                        df_tmin[Columns.DATE.value].max(),
+                    ]
+                )
+            else:
+                start_date = df[Columns.DATE.value].min()
+                end_date = df[Columns.DATE.value].max()
+
+        tz_delta = self._get_timedelta_from_timezones(timezone_, pytz.UTC)
+
+        df_tmean = pd.DataFrame(
+            {
+                Columns.DATE.value: pd.date_range(
+                    start_date + dt.timedelta(hours=tz_delta),
+                    end_date + dt.timedelta(hours=tz_delta),
+                    freq=self.stations.frequency.value,
+                    tz=pytz.UTC,
+                )
+            }
+        )
+
+        # merge both temperature dfs on dates
+        df_tmean = df_tmean.merge(df_tmax, how="left", on=Columns.DATE.value).merge(
+            df_tmin, how="left", on=Columns.DATE.value
+        )
+
+        df_tmean[Columns.VALUE.value] = (
+            df_tmean[tmax_key].astype(float, errors="ignore")
+            + df_tmean[tmin_key].astype(float, errors="ignore")
+        ) / 2
+
+        df_tmean = df_tmean.drop(columns=[tmax_key, tmin_key])
+
+        df_tmean[Columns.STATION_ID.value] = station_id
+        df_tmean[Columns.PARAMETER.value] = tmean_key
+        df_tmean[Columns.QUALITY.value] = pd.NA
+
+        df = df.append(df_tmean)
```

Comment:
this line is not required

Reply:
So I shouldn't calculate the mean temperature? Then what's the point? If the data is not appended, it is gone!

Reply:
No I mean that you can rewrite these two lines: 
```
return df.append(df_tmean)
```
It is just a minor comment to shorten the code.